apps/dashboard/src/pages/reputation.tsx

```diff
@@ -0,0 +1,58 @@
+import { Flex, Text } from "@chakra-ui/react"
+import { useEffect } from "react"
+import { useSearchParams } from "react-router-dom"
+import { addMemberByReputation, setOAuthState } from "../api/bandadaAPI"
+
+export default function ReputationPage() {
+    const [searchParams] = useSearchParams()
+
+    // Step 1
+    useEffect(() => {
```

Comment:
I think these can be one useEffect? Since there is dependency on searchParams, it would trigger when the params change. 

Also since its an async function inside the effect, we need to [cancel it](https://github.com/facebook/react/issues/14326) incase the component unmount.

Reply:
For the second point - if the component unmount before the async operation is settled, then it might result in unintended behaviours, and in some cases memory leaks. So ideally `useEffects` should have a "cleanup" function returned from it.

Something like this:
```
useEffect(() => {
  let didCancel = false;

  () => {
    const stateId = await setOAuthState(
                    groupId as string,
                    memberId as string,
                    "github",
                    redirectURI
                )
                
    if (!didCancel) { // Ignore if the user left the page
                window.location.replace(
                    `https://github.com/login/oauth/authorize?client_id=${
                        import.meta.env.VITE_GITHUB_CLIENT_ID
                    }&state=${stateId}`
                )
    }
  })();

  return () => { didCancel = true; }; 
}, []);
```

Reply:
Yea, true. I have seen it missing in other places as well.

Reply:
I'll open another issue for it then!